core/include/drivers/tzc400.h

```diff
@@ -138,15 +145,15 @@
 #define INT_CLEAR_CLEAR_SHIFT		0
 #define INT_CLEAR_CLEAR_MASK		0xf
 
-#define FAIL_CONTROL_DIR_SHIFT		(1 << 24)
-#define  FAIL_CONTROL_DIR_READ		0x0
-#define  FAIL_CONTROL_DIR_WRITE		0x1
-#define FAIL_CONTROL_NS_SHIFT		(1 << 21)
-#define  FAIL_CONTROL_NS_SECURE		0x0
-#define  FAIL_CONTROL_NS_NONSECURE	0x1
-#define FAIL_CONTROL_PRIV_SHIFT		(1 << 20)
-#define  FAIL_CONTROL_PRIV_PRIV		0x0
-#define  FAIL_CONTROL_PRIV_UNPRIV	0x1
+#define FAIL_CONTROL_DIR		BIT(24)
```

Comment:
How about defining it as:
``` C
/* If set write access, else read access */
#define FAIL_CONTROL_DIRECTION_WRITE BIT(24)
/* If set non-secure access, else secure access */
#define FAIL_CONTROL_NONSECURE       BIT(21)
/* If set privileged access, else unprivileged access */
#define FAIL_CONTROL_PRIVILEGED      BIT(20)
```

Easier to use since it only involves one define for checking each bit.

Reply:
much simpler. I will update. thanks.